tracing/src/main/java/com/palantir/tracing/DetachedSpan.java

```diff
@@ -68,6 +68,17 @@ static DetachedSpan start(
         return Tracer.detachInternal(observability, traceId, parentSpanId, operation, type);
     }
 
+    /**
+     * Creates a {@link DetachedSpan} instance based on the current tracing state without adding a new span.
+     * If there is no tracing state present a no-op instance is returned. This is equivalent to {@link #attach()}.
+     *
+     * @see DetachedSpan#attach()
+     */
+    @CheckReturnValue
+    static DetachedSpan detach() {
```

Comment:
It feels potentially wrong to return `DetachedSpan` here because it provides (several) methods to `complete()` the span, however these cannot be completed because they don't actually produce spans.

Perhaps we extract a super-interface from `DetachedSpan` called `Detached` or `DetachedContext` which provides all the detached-span methods except for completion?